File: is_prime.py
import sys

def is_prime(num):
	if num < 2:
		return False
	if num == 2:
		return True
	if num % 2 == 0:
		return False
	i = 3
	while i * i <= num:
		if num % i == 0:
			return False
		i += 2
	return True

def test_1():
	count = 0
	for i in range(1000):
		count += 1 if is_prime(i) else 0
	assert count == 168

if __name__ == '__main__':
	print(is_prime(int(sys.argv[1])))

File: is_prime.cpp
#include <iostream>

typedef unsigned long long ull;

bool is_prime(const ull num) {
	if (num < 2) {
		return false;
	}
	if (num == 2) {
		return true;
	}
	if ((num % 2) == 0) {
		return false;
	}
	ull i = 3;
	while (i * i <= num) {
		if (num % i == 0) {
			return false;
		}
		i += 2;
	}
	return true;
}

int main(int argc, char** argv) {
	std::cout << is_prime(std::stoull(argv[1])) << std::endl;
	return 0;
}
